module/VuFind/src/VuFind/View/Helper/Root/MakeLink.php

```diff
@@ -0,0 +1,109 @@
+<?php
+/**
+ * Make link view helper
+ *
+ * PHP version 7
+ *
+ * Copyright (C) Villanova University 2020.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+namespace VuFind\View\Helper\Root;
+
+use Laminas\View\Helper\AbstractHelper;
+
+/**
+ * Make link view helper
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class MakeLink extends AbstractHelper
+{
+    /**
+     * Create an anchor tag
+     *
+     * If $href, will try to find an href in $attrs
+     * > makeLink('text', ['href' => '#', 'class' => 'btn-link'])
+     *
+     * If $attrs is a string, will be treated like a class
+     * > makeLink('text', 'btn-link') == makeLink('text', ['class' => 'btn-link'])
+     *
+     * @param string       $text  Link contents
+     * @param string|array $href  Link destination (skippable for (text, attrs))
+     * @param array        $attrs Link attributes (associative array)
+     *
+     * @return string HTML for an anchor tag
+     */
+    public function __invoke($text, $href = null, $attrs = [])
+    {
+        // Class string
+        if (is_string($attrs)) {
+            $attrs = ['class' => $attrs];
+        }
+
+        // Skipped $href, treat $href like $attrs
+        if (is_array($href)) {
```

Comment:
Probably worth adding a test case for this scenario, regardless of how it is handled, so the expected behavior is defined.

Reply:
I see that you've refactored this somewhat, but I still think it's problematic (possibly now in new/different ways). For example, if you call it with the parameters `('my text', ['id' => 'my id'], 'myClass')` as parameters, I think something unexpected (and undesirable) is going to happen -- the special case for class strings is going to override the special case for arrays in the second parameter, and the ID attribute will get dropped. How important are these special cases? Can we simplify further to prevent confusion? Also note that the comment above will likely need more updating to match reality.

Reply:
In this case, 'myClass' would be completely ignored. Are you asking for a simplification of behavior (remove some flexibility) or a simplification of the code?

Reply:
You are, of course, correct -- I misinterpreted the code. I think it's okay as it is, but the example in lines 52-53 is not quite right and should be adjusted.

Reply:
To be honest, if you're worried about confusion, I can see the argument for removing the $href skip flexibility. It really only saves 5 characters as a short cut.

Reply:
Yeah, that might be worth doing, just to improve code readability. We can always add it back in the future if there's a really compelling reason to.